app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java

```diff
@@ -56,6 +62,8 @@
     @InjectView(R.id.patient_sex_female) RadioButton mSexFemale;
     @InjectView(R.id.patient_sex_male) RadioButton mSexMale;
 
+    private static final Pattern ID_PATTERN = Pattern.compile("([a-zA-Z]+)/?([0-9]+)*");
```

Comment:
It could also result in IndexOutOfBoundsExceptions, because it makes the second subcapture optional, and the code assumes that the second subcapture is available.

As Ping states, I think you're after `([a-zA-Z]+)/?([0-9]*)` instead.